[Bug Report]: Something_went_wrong_while_trying_to_build_project (VSC-643)

**Describe the bug**
Upgrading to 1.0.3 made the component dead.
At the same time, idf.py in the project folder does its job correctly.
Rolling back to version 1.0.2 makes the component healthy again.

Windows 10 + VS-Code, ESP IDF preinstalled, Python version 3.8.7 used in system
Something went wrong while trying to build project

**Screenshots**
If applicable, add screenshots to help explain your problem.
http://i.piccy.info/i9/04ff4594c4420976645cbc55f404ccd8/1619605350/24924/1426873/Something_went_wrong_while_trying_to_build_project.png
**Environment (please complete the following information):**

- OS Name & Version: [e.g. MacOS X 10.14.5]
- VSCode Version: [e.g. 1.35.1]
- ESP-IDF Version: [e.g. v3.2.2]
- Python Version: [e.g. 2.7.10]

> **NOTE**: You can use the `ESP-IDF: Doctor command` to generate a report of your configuration.
---------------------------------------------- ESP-IDF Extension for Visual Studio Code report ---------------------------------------------
OS win32 x64 10.0.19042 
System environment variable PATH 
 C:\dev\Python38\Scripts\;C:\dev\Python38\;C:\Program Files\Common Files\Oracle\Java\javapath;C:\windows\system32;C:\windows;C:\windows\System32\Wbem;C:\windows\System32\WindowsPowerShell\v1.0\;C:\windows\System32\OpenSSH\;C:\dev\Git\Git\cmd;C:\dev\CMake\bin;C:\Program Files\PuTTY\;C:\dev\Python38\Scripts\;C:\dev\Python38\;C:\Users\Slava\AppData\Local\Microsoft\WindowsApps;C:\Users\Slava\AppData\Local\Programs\Microsoft VS Code\bin; 
Visual Studio Code version 1.55.2 
Visual Studio Code language en 
Visual Studio Code shell C:\windows\System32\cmd.exe 
ESP-IDF Extension version 1.0.3 
---------------------------------------------------- Extension configuration settings ------------------------------------------------------
ESP-IDF Path (idf.espIdfPath) c:\dev\espidf42
Custom extra paths (idf.customExtraPaths) C:\dev\espidftools\.espressif\tools\xtensa-esp32-elf\esp-2020r3-8.4.0\xtensa-esp32-elf\bin;C:\dev\espidftools\.espressif\tools\xtensa-esp32s2-elf\esp-2020r3-8.4.0\xtensa-esp32s2-elf\bin;C:\dev\espidftools\.espressif\tools\esp32ulp-elf\2.28.51-esp-20191205\esp32ulp-elf-binutils\bin;C:\dev\espidftools\.espressif\tools\esp32s2ulp-elf\2.28.51-esp-20191205\esp32s2ulp-elf-binutils\bin;C:\dev\espidftools\.espressif\tools\cmake\3.16.4\bin;C:\dev\espidftools\.espressif\tools\openocd-esp32\v0.10.0-esp32-20200709\openocd-esp32\bin;C:\dev\espidftools\.espressif\tools\ninja\1.10.0;C:\dev\espidftools\.espressif\tools\idf-exe\1.0.1;C:\dev\espidftools\.espressif\tools\ccache\3.7;C:\dev\espidftools\.espressif\tools\dfu-util\0.9\dfu-util-0.9-win64
Custom extra vars (idf.customExtraVars) {"OPENOCD_SCRIPTS":"C:\\dev\\espidftools\\.espressif\\tools\\openocd-esp32\\v0.10.0-esp32-20200709/openocd-esp32/share/openocd/scripts","IDF_CCACHE_ENABLE":"1"}
Virtual env Python Path (idf.pythonBinPath) C:\dev\espidftools\.espressif\python_env\idf4.2_py3.8_env\Scripts\python.exe
Serial port (idf.port) /dev/ttyUSB1
OpenOCD Configs (idf.openOcdConfigs) interface/ftdi/esp32_devkitj_v1.cfg,board/esp32-wrover.cfg
ESP-IDF Tools Path (idf.toolsPath) ${env:IDF_TOOLS_PATH}
-------------------------------------------------------- Configurations access -------------------------------------------------------------
Access to ESP-IDF Path (idf.espIdfPath) true
Access to ESP-IDF Custom extra paths
Access to C:\dev\espidftools\.espressif\tools\xtensa-esp32-elf\esp-2020r3-8.4.0\xtensa-esp32-elf\bin: true
Access to C:\dev\espidftools\.espressif\tools\xtensa-esp32s2-elf\esp-2020r3-8.4.0\xtensa-esp32s2-elf\bin: true
Access to C:\dev\espidftools\.espressif\tools\esp32ulp-elf\2.28.51-esp-20191205\esp32ulp-elf-binutils\bin: true
Access to C:\dev\espidftools\.espressif\tools\esp32s2ulp-elf\2.28.51-esp-20191205\esp32s2ulp-elf-binutils\bin: true
Access to C:\dev\espidftools\.espressif\tools\cmake\3.16.4\bin: true
Access to C:\dev\espidftools\.espressif\tools\openocd-esp32\v0.10.0-esp32-20200709\openocd-esp32\bin: true
Access to C:\dev\espidftools\.espressif\tools\ninja\1.10.0: true
Access to C:\dev\espidftools\.espressif\tools\idf-exe\1.0.1: true
Access to C:\dev\espidftools\.espressif\tools\ccache\3.7: true
Access to C:\dev\espidftools\.espressif\tools\dfu-util\0.9\dfu-util-0.9-win64: true
Access to Virtual env Python Path (idf.pythonBinPath) true
Access to CMake in environment PATH undefined
Access to Ninja in environment PATH undefined
Access to ESP-IDF Tools Path (idf.toolsPath) false
----------------------------------------------------------- Executables Versions -----------------------------------------------------------
Git version 2.31.1.windows.1
ESP-IDF version 4.2
Python version 3.8.7
Python's pip version 20.3.3
-------------------------------------------------- Python packages in idf.pythonBinPath ----------------------------------------------------
bidict version: 0.21.2
bitstring version: 3.1.7
Brotli version: 1.0.9
certifi version: 2020.12.5
cffi version: 1.14.4
chardet version: 4.0.0
click version: 7.1.2
cryptography version: 3.3.1
ecdsa version: 0.16.1
esp-windows-curses version: 0.1
Flask version: 0.12.5
Flask-Compress version: 1.8.0
Flask-SocketIO version: 2.9.6
future version: 0.18.2
gcovr version: 4.2
gdbgui version: 0.13.2.0
gevent version: 1.5.0
greenlet version: 1.0.0
idna version: 2.10
itsdangerous version: 1.1.0
Jinja2 version: 2.11.2
lxml version: 4.6.3
MarkupSafe version: 1.1.1
pip version: 20.3.3
psutil version: 5.8.0
pycparser version: 2.20
pyelftools version: 0.27
pygdbmi version: 0.9.0.2
Pygments version: 2.7.4
pyparsing version: 2.3.1
pyserial version: 3.5
python-engineio version: 4.0.0
python-socketio version: 5.0.4
pywin32 version: 300
reedsolo version: 1.5.4
requests version: 2.25.1
setuptools version: 51.3.3
six version: 1.15.0
urllib3 version: 1.26.4
websocket-client version: 0.58.0
Werkzeug version: 0.16.1
wheel version: 0.36.2
windows-curses version: 2.2.0
xmlrunner version: 1.7.7
---------------------------------------------------- Check ESP-IDF python requirements.txt -------------------------------------------------
Check ESP-IDF Python packages Python requirements from c:\dev\espidf42\requirements.txt are satisfied.
---------------------------------------------------- Check extension requirements.txt ------------------------------------------------------
Check Extension Python packages undefined
---------------------------------------------------- Check ESP-IDF debug adapter requirements.txt ------------------------------------------
Check Debug AdapterPython packages undefined
----------------------------------------------------------- Latest error -----------------------------------------------------------------
Latest error at Command failed: C:\dev\espidftools\.espressif\python_env\idf4.2_py3.8_env\Scripts\python.exe c:\dev\espidf42\tools\check_python_dependencies.py -r "c:\Users\Slava\.vscode\extensions\espressif.esp-idf-extension-1.0.3\requirements.txt"

--------------------------------------------------------------------------------------------------------------------------------------------

**Logs**

- Windows: `%USERPROFILE%\.vscode\extensions\espressif.esp-idf-extension-VERSION\esp_idf_vsc_ext.log`
{"user":true,"message":"Something went wrong while trying to build the project","stack":"Error: CMake or Ninja executables not found\n\tat d.<anonymous> (c:\\Users\\Slava\\.vscode\\extensions\\espressif.esp-idf-extension-1.0.3\\dist\\extension.js:1:110335)\n\tat Generator.next (<anonymous>)\n\tat s (c:\\Users\\Slava\\.vscode\\extensions\\espressif.esp-idf-extension-1.0.3\\dist\\extension.js:1:109368)\n\tat processTicksAndRejections (internal/process/task_queues.js:97:5)","level":"error","timestamp":"2021-04-28T09:37:22.729Z"}
{"message":"non zero exit code 1\r\n\r\n����������: �� 㤠���� ���� 䠩�� �� ������� 蠡�����.\r\n","stack":"Error: non zero exit code 1\r\n\r\n����������: �� 㤠���� ���� 䠩�� �� ������� 蠡�����.\r\n\n\tat ChildProcess.<anonymous> (c:\\Users\\Slava\\.vscode\\extensions\\espressif.esp-idf-extension-1.0.3\\dist\\extension.js:1:5604)\n\tat ChildProcess.emit (events.js:315:20)\n\tat Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)","level":"error","timestamp":"2021-04-28T09:38:09.315Z"}
{"message":"Cannot access filePath: ninja","level":"error","timestamp":"2021-04-28T09:38:09.315Z"}
{"user":true,"message":"Something went wrong while trying to build the project","stack":"Error: CMake or Ninja executables not found\n\tat d.<anonymous> (c:\\Users\\Slava\\.vscode\\extensions\\espressif.esp-idf-extension-1.0.3\\dist\\extension.js:1:110335)\n\tat Generator.next (<anonymous>)\n\tat s (c:\\Users\\Slava\\.vscode\\extensions\\espressif.esp-idf-extension-1.0.3\\dist\\extension.js:1:109368)\n\tat processTicksAndRejections (internal/process/task_queues.js:97:5)","level":"error","timestamp":"2021-04-28T09:38:09.316Z"}
{"message":"non zero exit code 1\r\n\r\n����������: �� 㤠���� ���� 䠩�� �� ������� 蠡�����.\r\n","stack":"Error: non zero exit code 1\r\n\r\n����������: �� 㤠���� ���� 䠩�� �� ������� 蠡�����.\r\n\n\tat ChildProcess.<anonymous> (c:\\Users\\Slava\\.vscode\\extensions\\espressif.esp-idf-extension-1.0.3\\dist\\extension.js:1:5604)\n\tat ChildProcess.emit (events.js:315:20)\n\tat Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)","level":"error","timestamp":"2021-04-28T09:39:32.147Z"}
{"message":"Cannot access filePath: ninja","level":"error","timestamp":"2021-04-28T09:39:32.148Z"}

**Extension Configuration settings**
Open Command Palette (F1) and type `Preferences: Open Settings (JSON)` or use `{CURRENT_PROJECT}/.vscode/settings.json` to get the following extension configuration settings value:
settings.json
{
  "files.associations": {
    "stdio.h": "c"
  }
}

The tips that I applied - killed the pre-installed ESP IDF. Now compilation only works from VSCode version 1.0.2 and does not work from terminal. Also reinstall IDF.
I can only advise everyone not to try to install version 1.0.3 in any case and stay on 1.0.2
//-------------
C:\dev\esp32workspace\get-started\blink>idf.py build
Executing action: all (aliases: build)
Running cmake in directory c:\dev\esp32workspace\get-started\blink\build
Executing "cmake -G Ninja -DPYTHON_DEPS_CHECKED=1 -DESP_PLATFORM=1 -DIDF_TARGET=esp32 -DCCACHE_ENABLE=1 c:\dev\esp32workspace\get-started\blink"...
-- Found Git: C:/dev/Git/Git/bin/git.exe (found version "2.31.1.windows.1")
-- ccache will be used for faster recompilation
-- The C compiler identification is GNU 8.4.0
-- The CXX compiler identification is GNU 8.4.0
-- The ASM compiler identification is GNU
-- Found assembler: C:/dev/espidftools/.espressif/tools/xtensa-esp32-elf/esp-2020r3-8.4.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-gcc.exe
-- Check for working C compiler: C:/dev/espidftools/.espressif/tools/xtensa-esp32-elf/esp-2020r3-8.4.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-gcc.exe
-- Check for working C compiler: C:/dev/espidftools/.espressif/tools/xtensa-esp32-elf/esp-2020r3-8.4.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-gcc.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: C:/dev/espidftools/.espressif/tools/xtensa-esp32-elf/esp-2020r3-8.4.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-g++.exe
-- Check for working CXX compiler: C:/dev/espidftools/.espressif/tools/xtensa-esp32-elf/esp-2020r3-8.4.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-g++.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Project is not inside a git repository, or git repository has no commits; will not use 'git describe' to determine PROJECT_VER.
-- Building ESP-IDF components for target esp32
Cannot parse CMake requirements line: __component_set_property(___idf_Terminalengine REQUIRES "i2Cengine;Thermostat;userinit;cxx;newlib;freertos;heap;log;lwip;soc;esp_rom;esp_common;esp_system;xtensa;esp32")

CMake Error at C:/dev/espidf42/tools/cmake/component.cmake:240 (message):
Call Stack (most recent call first):
  C:/dev/espidf42/tools/cmake/build.cmake:405 (__component_get_requirements)
  C:/dev/espidf42/tools/cmake/project.cmake:395 (idf_build_process)
  CMakeLists.txt:7 (project)


-- Configuring incomplete, errors occurred!
See also "C:/dev/esp32workspace/get-started/blink/build/CMakeFiles/CMakeOutput.log".
cmake failed with exit code 1

C:\dev\esp32workspace\get-started\blink>

--- a/docs/SETTINGS.md
+++ b/docs/SETTINGS.md
@@ -8,16 +8,17 @@ This extension contributes the following settings that can be later updated in s
 
 These are the configuration settings that ESP-IDF extension contributes to your Visual Studio Code editor settings.
 
-| Setting ID             | Description                                                                   |
-| ---------------------- | ----------------------------------------------------------------------------- |
-| `idf.customExtraPaths` | Paths to be appended to \$PATH                                                |
-| `idf.customExtraVars`  | Variables to be added to system environment variables                         |
-| `idf.espIdfPath`       | Path to locate ESP-IDF framework (IDF_PATH)                                   |
-| `idf.espIdfPathWin`    | Path to locate ESP-IDF framework in Windows (IDF_PATH)                        |
-| `idf.pythonBinPath`    | Python absolute binary path used to execute ESP-IDF Python Scripts            |
-| `idf.pythonBinPathWin` | Python absolute binary path used to execute ESP-IDF Python Scripts in Windows |
-| `idf.toolsPath`        | Path to locate ESP-IDF Tools (IDF_TOOLS_PATH)                                 |
-| `idf.toolsPathWin`     | Path to locate ESP-IDF Tools in Windows (IDF_TOOLS_PATH)                      |
+| Setting ID                      | Description                                                                   |
+| ------------------------------- | ----------------------------------------------------------------------------- |
+| `idf.customExtraPaths`          | Paths to be appended to \$PATH                                                |
+| `idf.customExtraVars`           | Variables to be added to system environment variables                         |
+| `idf.espIdfPath`                | Path to locate ESP-IDF framework (IDF_PATH)                                   |
+| `idf.espIdfPathWin`             | Path to locate ESP-IDF framework in Windows (IDF_PATH)                        |
+| `idf.pythonBinPath`             | Python absolute binary path used to execute ESP-IDF Python Scripts            |
+| `idf.pythonBinPathWin`          | Python absolute binary path used to execute ESP-IDF Python Scripts in Windows |
+| `idf.toolsPath`                 | Path to locate ESP-IDF Tools (IDF_TOOLS_PATH)                                 |
+| `idf.toolsPathWin`              | Path to locate ESP-IDF Tools in Windows (IDF_TOOLS_PATH)                      |
+| `idf.enableIdfComponentManager` | Enable IDF Component manager in build command                                 |
 
 This is how the extension uses them:
 

--- a/i18n/en/package.i18n.json
+++ b/i18n/en/package.i18n.json
@@ -78,6 +78,7 @@
   "param.saveScope": "Where to save configuration with ESP-IDF commands with number value as vscode.ConfigurationTarget. Global = 1, Workspace= 2, WorkspaceFolder=3",
   "param.rainmaker.api.server_url": "ESP-Rainmaker cloud server URL",
   "param.launchMonitorOnDebugSession.title": "Start IDF Monitor along with ESP-IDF Debug Adapter session",
+  "param.enableIdfComponentManager.title": "Enable IDF Component Manager in build task",
   "esp.rainmaker.backend.sync.title": "Sync with ESP-Rainmaker Cloud Server",
   "esp.rainmaker.backend.connect.title": "Connect with ESP-Rainmaker Cloud Server",
   "esp.rainmaker.backend.logout.title": "Unlink Rainmaker Account",

--- a/i18n/es/package.i18n.json
+++ b/i18n/es/package.i18n.json
@@ -64,6 +64,7 @@
   "param.uncoveredLightTheme": "Color de fondo para lineas no cubiertas en temas claros por ESP-IDF Coverage.",
   "param.uncoveredDarkTheme": "Color de fondo para lineas no cubiertas en temas oscuros por ESP-IDF Coverage.",
   "param.launchMonitorOnDebugSession.title": "Iniciar IDF Monitor junto a la sesión ESP-IDF Debug Adapter",
+  "param.enableIdfComponentManager.title": "Habilitar IDF Component Manager en construccion de proyecto",
   "view.components.name": "Componentes de proyecto",
   "configuration.title": "ESP-IDF",
   "espIdf.apptrace.archive.refresh.title": "Refresh Trace Archive List",

--- a/i18n/ru/package.i18n.json
+++ b/i18n/ru/package.i18n.json
@@ -64,6 +64,7 @@
   "param.uncoveredLightTheme": "Цвет фона для непокрытых линий в светлой теме для покрытия кода ESP-IDF.",
   "param.uncoveredDarkTheme": "Цвет фона для непокрытых линий в темной теме для покрытия кода ESP-IDF.",
   "param.launchMonitorOnDebugSession.title": "Запустите IDF Monitor вместе с сеансом ESP-IDF Debug Adapter.",
+  "param.enableIdfComponentManager.title": "Включить диспетчер компонентов IDF в задаче сборки",
   "view.components.name": "Компоненты проекта",
   "configuration.title": "ESP-IDF",
   "espIdf.apptrace.archive.refresh.title": "Обновить список архива трассировки",

--- a/i18n/zh-CN/package.i18n.json
+++ b/i18n/zh-CN/package.i18n.json
@@ -64,6 +64,7 @@
   "param.uncoveredLightTheme": "ESP-IDF覆盖的光主题中未覆盖线条的背景色",
   "param.uncoveredDarkTheme": "ESP-IDF覆盖的深色主题中未覆盖线条的背景色",
   "param.launchMonitorOnDebugSession.title": "启动IDF监视器和ESP-IDF调试适配器会话",
+  "param.enableIdfComponentManager.title": "在生成任务中启用IDF组件管理器",
   "view.components.name": "项目组件",
   "configuration.title": "ESP-IDF",
   "espIdf.apptrace.archive.refresh.title": "刷新跟踪归档列表",

--- a/package.json
+++ b/package.json
@@ -617,6 +617,11 @@
             "description": "%param.launchMonitorOnDebugSession.title%",
             "scope": "resource",
             "default": false
+          },
+          "idf.enableIdfComponentManager": {
+            "type": "boolean",
+            "description": "%param.enableIdfComponentManager.title%",
+            "default": false
           }
         }
       }

--- a/package.nls.json
+++ b/package.nls.json
@@ -78,6 +78,7 @@
   "param.saveScope": "Where to save configuration with ESP-IDF commands with number value as vscode.ConfigurationTarget. Global = 1, Workspace= 2, WorkspaceFolder=3",
   "param.rainmaker.api.server_url": "ESP-Rainmaker cloud server URL",
   "param.launchMonitorOnDebugSession.title": "Start IDF Monitor along with ESP-IDF Debug Adapter session",
+  "param.enableIdfComponentManager.title": "Enable IDF Component Manager in build task",
   "esp.rainmaker.backend.sync.title": "Sync with ESP-Rainmaker Cloud Server",
   "esp.rainmaker.backend.connect.title": "Connect with ESP-Rainmaker Cloud Server",
   "esp.rainmaker.backend.logout.title": "Unlink Rainmaker Account",

--- a/src/utils.ts
+++ b/src/utils.ts
@@ -776,7 +776,13 @@ export function appendIdfAndToolsToPath() {
   }
   modifiedEnv.IDF_TARGET = idfTarget || process.env.IDF_TARGET;
 
-  modifiedEnv.IDF_COMPONENT_MANAGER = "1";
+  let enableComponentManager = idfConf.readParameter(
+    "idf.enableIdfComponentManager"
+  ) as boolean;
+
+  if (enableComponentManager) {
+    modifiedEnv.IDF_COMPONENT_MANAGER = "1";
+  }
 
   return modifiedEnv;
 }

